drivers/rkt/driver.go

```diff
@@ -0,0 +1,996 @@
+	"golang.org/x/net/context"
+
+	"encoding/json"
+	"io/ioutil"
+	"math/rand"
+	"strconv"
+
+	"github.com/hashicorp/consul-template/signals"
+	"github.com/hashicorp/nomad/client/driver/executor"
+)
+
+const (
+	// pluginName is the name of the plugin
+	pluginName = "rkt"
+
+	// fingerprintPeriod is the interval at which the driver will send fingerprint responses
+	fingerprintPeriod = 30 * time.Second
+
+	// minRktVersion is the earliest supported version of rkt. rkt added support
+	// for CPU and memory isolators in 0.14.0. We cannot support an earlier
+	// version to maintain an uniform interface across all drivers
+	minRktVersion = "1.27.0"
+
+	// rktCmd is the command rkt is installed as.
+	rktCmd = "rkt"
+
+	// rktNetworkDeadline is how long to wait for container network to start
+	rktNetworkDeadline = 1 * time.Minute
+)
+
+var (
+	// pluginInfo is the response returned for the PluginInfo RPC
+	pluginInfo = &base.PluginInfoResponse{
+		Type:             base.PluginTypeDriver,
+		PluginApiVersion: "0.0.1",
+		PluginVersion:    "0.1.0",
+		Name:             pluginName,
+	}
+
+	// configSpec is the hcl specification returned by the ConfigSchema RPC
+	configSpec = hclspec.NewObject(map[string]*hclspec.Spec{
+		"volumes_enabled": hclspec.NewDefault(
+			hclspec.NewAttr("volumes_enabled", "bool", false),
+			hclspec.NewLiteral("true"),
+		),
+	})
+
+	// taskConfigSpec is the hcl specification for the driver config section of
+	// a taskConfig within a job. It is returned in the TaskConfigSchema RPC
+	taskConfigSpec = hclspec.NewObject(map[string]*hclspec.Spec{
+		"image":              hclspec.NewAttr("image", "string", true),
+		"command":            hclspec.NewAttr("command", "string", false),
+		"args":               hclspec.NewAttr("args", "list(string)", false),
+		"trust_prefix":       hclspec.NewAttr("trust_prefix", "string", false),
+		"dns_servers":        hclspec.NewAttr("dns_servers", "list(string)", false),
+		"dns_search_domains": hclspec.NewAttr("dns_search_domains", "list(string)", false),
+		"net":                hclspec.NewAttr("net", "list(string)", false),
+		"port_map":           hclspec.NewAttr("port_map", "map(string)", false),
+		"volumes":            hclspec.NewAttr("volumes", "list(string)", false),
+		"insecure_options":   hclspec.NewAttr("insecure_options", "list(string)", false),
+		"no_overlay":         hclspec.NewAttr("no_overlay", "bool", false),
+		"debug":              hclspec.NewAttr("debug", "bool", false),
+		"group":              hclspec.NewAttr("group", "string", false),
+	})
+
+	// capabilities is returned by the Capabilities RPC and indicates what
+	// optional features this driver supports
+	capabilities = &drivers.Capabilities{
+		SendSignals: true,
+		Exec:        true,
+		FSIsolation: cstructs.FSIsolationChroot,
+	}
+
+	reRktVersion  = regexp.MustCompile(`rkt [vV]ersion[:]? (\d[.\d]+)`)
+	reAppcVersion = regexp.MustCompile(`appc [vV]ersion[:]? (\d[.\d]+)`)
+
+	_ drivers.DriverPlugin = (*RktDriver)(nil)
```

Comment:
I put it here because it was useful during development as I implemented methods in the interface. Don't have a strong opinion, can move it to the test too